internal/provider/datasource_directory_roles_test.go

```diff
@@ -26,6 +26,7 @@ func TestDataSourceDirectoryRoles(t *testing.T) {
 					Check: resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr("data.btp_directory_roles.uut", "directory_id", "05368777-4934-41e8-9f3c-6ec5f4d564b9"),
 						resource.TestCheckResourceAttr("data.btp_directory_roles.uut", "values.#", "8"),
+						resource.TestMatchResourceAttr("data.btp_directory_roles.uut", "values.app_name", regexp.MustCompile(`(.*?)`)),
```

Comment:
I assume this cannot fail since the index is missing. The key should look like "values.0.app_name". Probably, this regex is always satisfied. Maybe it is enough to check that the app_name is set or that is not empty.

Reply:
Weird that it also does not care about the index at all. I added the first entry now to give some basic check,